client/src/components/editor/LinkSourceAnet.js

```diff
@@ -0,0 +1,245 @@
+import {
+  ReportDetailedOverlayRow,
+  OrganizationOverlayRow,
+  TaskSimpleOverlayRow,
+  PersonDetailedOverlayRow,
+  PositionOverlayRow,
+  LocationOverlayRow
+} from "components/advancedSelectWidget/AdvancedSelectOverlayRow"
+
+import ButtonToggleGroup from "components/ButtonToggleGroup"
+import { ENTITY_TYPES } from "utils_links"
+import pluralize from "pluralize"
+import createEntity from "./utils/createEntity"
+
+const entityFilters = {
+  allEntities: {
+    label: "All entities",
+    queryVars: {}
+  }
+}
+
+const peopleFilters = {
+  allEntities: {
+    label: "All",
+    queryVars: { matchPositionName: true }
+  },
+  activeAdvisors: {
+    label: "All advisors",
+    queryVars: { role: Models.Person.ROLE.ADVISOR, matchPositionName: true }
+  },
+  activePrincipals: {
+    label: "All principals",
+    queryVars: { role: Models.Person.ROLE.PRINCIPAL }
+  }
+}
+
+const widgetPropsReport = {
+  objectType: Models.Report,
+  overlayRenderRow: ReportDetailedOverlayRow,
+  overlayColumns: ["Goal", "Author", "Updated"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Report.autocompleteQuery,
+  addon: REPORTS_ICON
+}
+
+const widgetPropsPeople = {
+  objectType: Models.Person,
+  overlayRenderRow: PersonDetailedOverlayRow,
+  overlayColumns: ["Name", "Position", "Location", "Organization"],
+  filterDefs: peopleFilters,
+  queryParams: {},
+  fields: Models.Person.autocompleteQuery,
+  addon: PEOPLE_ICON
+}
+
+const widgetPropsOrganization = {
+  objectType: Models.Organization,
+  overlayRenderRow: OrganizationOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Organization.autocompleteQuery,
+  addon: ORGANIZATIONS_ICON
+}
+
+const widgetPropsPosition = {
+  objectType: Models.Position,
+  overlayRenderRow: PositionOverlayRow,
+  overlayColumns: ["Position", "Organization", "Current Occupant"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Position.autocompleteQuery,
+  addon: POSITIONS_ICON
+}
+
+const widgetPropsLocation = {
+  objectType: Models.Location,
+  overlayRenderRow: LocationOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: { status: Models.Location.STATUS.ACTIVE },
+  fields: Models.Location.autocompleteQuery,
+  addon: LOCATIONS_ICON
+}
+
+const widgetPropsTask = {
+  objectType: Models.Task,
+  overlayRenderRow: TaskSimpleOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: { status: Models.Task.STATUS.ACTIVE },
+  fields: Models.Task.autocompleteQuery,
+  addon: TASKS_ICON
+}
+
+const widgetTypeMapping = {
+  [ENTITY_TYPES.REPORT]: widgetPropsReport,
+  [ENTITY_TYPES.PERSON]: widgetPropsPeople,
+  [ENTITY_TYPES.ORGANIZATION]: widgetPropsOrganization,
+  [ENTITY_TYPES.POSITION]: widgetPropsPosition,
+  [ENTITY_TYPES.LOCATION]: widgetPropsLocation,
+  [ENTITY_TYPES.TASK]: widgetPropsTask
+}
+
+class LinkSourceAnet extends Component {
```

Comment:
This really ought to be a pure functional component.

Reply:
Indeed! I already changed the rest, but did not want to rework 'old work'. However, since I am busy I will also convert this one.